low-level/sensors/impl/generic/raid_integrity_data.py

```diff
@@ -166,22 +165,27 @@ def _raid_health_monitor(self):
                 # Check RAID device array state is 'idle' or not
                 result = self._check_raid_state(device)
                 if result == "failed":
+                    # Send alert
+                    self.alert_type = self.FAULT
+                    self._alert_msg = RaidAlertMsgs.STATE_MSG.value
+                    self._send_json_msg(self.alert_type, device, self._alert_msg)
+                    # Retry to check RAID state
                     self._retry_execution(self._check_raid_state, device)
                 logger.debug("'idle' state is found in Raid device:{}."
                              .format(device))
 
                 # Check Mismatch count in RAID device files.
                 result = self._check_mismatch_count(device)
                 if result == "failed":
+                    # Send alert
+                    self.alert_type = self.FAULT
```

Comment:
fault & fault resolved needs to maintained & persisted(to handle sspl/node reboot scenarios) per device in mdraid array, or we are mixing up the status.

Reply:
Done.
I have kept a file to persist the fault state per device.